torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1296,8 +1296,8 @@ def make_singular_matrix_batch_base(size, rank):
 
         with torch.no_grad():
             n = size[-1]
-            a = make_arg(size[:-2] + (n, rank)) / 10
-            b = make_arg(size[:-2] + (rank, n)) / 10
+            a = make_arg(size[:-2] + (n, rank), dtype=torch.double) / 10
```

Comment:
This use of double is interesting because it overrides the dtype requested for the sample inputs. Is it necessary and will the resulting sample inputs be double because of it?

cc @Lezcano